imap/imapd.c

```diff
@@ -7910,6 +7910,10 @@ static void cmd_rename(char *tag, char *oldname, char *newname, char *location)
     if (!r && rename_user) {
         user_copyquotaroot(oldmailboxname, newmailboxname);
         user_renameacl(&imapd_namespace, newmailboxname, olduser, newuser);
+        if (config_getswitch(IMAPOPT_MAILBOX_LEGACY_DIRS)) {
```

Comment:
Hmmm.  Do we also need to check for MBTYPE_LEGACY_DIRS here?  Like, maybe the server is configured to use uuid-mailboxes, but this particular user hasn't been converted yet.  I'm not sure what the between-states look like but it feels important to catch them, if they can exist.

Reply:
Good point.  Fixed